x-pack/plugins/task_manager/server/task_running/task_runner.ts

```diff
@@ -734,6 +734,8 @@ export class TaskManagerRunner implements TaskRunner {
   ): Promise<Result<SuccessfulRunResult, FailedRunResult>> {
     const { task } = this.instance;
 
+    const taskHasExpired = this.isExpired;
```

Comment:
I'm decoupling task success/failure from task expired/finished on time since we don't yet require tasks to implement task cancellation so task manager may try to cancel a task but it will still run to completion successfully.